pytrip/tripexecuter2/plan.py

```diff
@@ -0,0 +1,293 @@
+        self.iterations = 500
+        self.opt_method = "phys"
+        self.opt_principle = "H2Obased"
+        self.opt_alg = "cg"
+        self.dose_alg = "cl"
+        self.bio_alg = "cl"
+        self.eps = 1e-3
+        self.geps = 1e-4
+
+        # algorithms for dose optimizations
+        self._opt_principles = {"H2Obased": "Very simplified single field optimization",
+                                "CTbased": "Full optimization, multiple fields"}
+
+        self._opt_methods = {"phys": "Physical dose only",
+                             "bio": "Biological optimization"}
+
+        self._opt_algs = {"cl": "classic",
+                          "cg": "conjugate graients (default)",
+                          "gr": "plain gradients",
+                          "bf": "Bortfeld's algorithm",
+                          "fr": "Fletcher-Reeves algorithm"}
+
+        self._bio_algs = {"cl": "classic (default)",
+                          "ld": "lowdose (faster)"}
+
+        self._dose_algs = {"cl": "classic (default)",
+                           "ap": "allpoints",
+                           "ms": "multiple scatter"}
+
+        # Scancap parameters:
+        # Thickness of ripple filter (0.0 for none)
+        # will only used for the TRiP98 Scancap command.
+        self.rifi = 0.0  # rifi thickness in [mm]
+        self.bolus = 0.0  # amount of bolus to be applied in [mm]
+        self.offh2o = 0.0  # Offset of exit-window and ionchambers in [mm]
+        self.minparticles = 5000  # smallest amount of particle which will go into a raster spot
+        self.scanpath = "none"
+        self._scanpaths = {"none": "No path, output as is",
+                           "uw": "U. Weber, efficient",
+                           "uw2": "very efficient, works also for non-grid points",
+                           "mk": "M. Kraemer, conservative"}
+
+        self.random_seed1 = 1000  # Only needed for biological optimization.
+        # TODO: TRiP may accept a second "seed2" depending on PRNG. Not implemented here.
+
+        # results requested from optimization
+        self.want_phys_dose = True
+        self.want_bio_dose = False
+        self.want_dlet = False
+        self.want_rst = False
+        self.want_tlet = False  # TODO: not implemented.
+
+        self.window = []  # window [xmin,xmax,ymin,ymax,zmin,zmax] in CTcoords [mm] for limited dose output.
+        self.target_dose = 2.0  # target dose in Gray
+        self.target_dose_percent = 100.0  # target dose in percent
+        self.target_dose_cube = None  # pass a DosCube() here, for incube() optimization.
+
+        self.target_tissue_type = ""  # target tissue type, for biological optimization
+        self.res_tissue_type = ""  # residual tissue types
+        self.incube = False  # enable or disable incube optimizations
+
+    def __str__(self):
+        """ Pretty print all attributes
```

Comment:
This method is way to long. What about exporting its functionality outside to 
 - private method in a module which takes plan as an argument and returns a string
 - helper class with static method which does the same 
?